staticd/static_zebra.c

```diff
@@ -201,6 +201,9 @@ static int static_zebra_nexthop_update(ZAPI_CALLBACK_ARGS)
 	if (nhr.prefix.family == AF_INET6)
 		afi = AFI_IP6;
 
+	if (CHECK_FLAG(nhr.flags, ZEBRA_FLAG_NHT_LOCAL_ADDR))
```

Comment:
Why isn't `nhr.type == ZEBRA_ROUTE_CONNECT` sufficient here?  Why do we need a flag?

Reply:
Hi @donaldsharp , 
It need to differentiate b/w valid connected next hop and local connected address as nexthop. 
Lets say: 
static route 1 :  ip route a.b.c.d/mask   `<valid` next hop addr>
static route 2 : ip route p.q.r.s/mask  `<local` connected address as nexhtop>
For both of these static route cases , type is populated as 'ZEBRA_ROUTE_CONNECT'.  where Static route-1 is valid config and static route-2 is not a valid configuration. 
so I have used a flag to differentiate this  and make it fail for next hop address as local connected address.

Reply:
why won't nhr.type = ZEBRA_ROUTE_CONNECT work?  Additionally this is not a nexthop tracking problem it is a staticd problem( and frankly staticd has all the information already ).  we should not be solving this in zebra.

Reply:
@donaldsharp 
As suggested, I have made this change specific to staticd.